soda-core/src/soda_core/common/soda_cloud.py

```diff
@@ -536,108 +553,101 @@ def verify_contract_on_agent(
         scan_id: str = response_json.get("scanId")
 
         log_records: Optional[list[LogRecord]] = None
-        if response.status_code == 200 and scan_id:
-            scan_is_finished, contract_dataset_cloud_url = self._poll_remote_scan_finished(
-                scan_id=scan_id, blocking_timeout_in_minutes=blocking_timeout_in_minutes
-            )
+        if response.status_code != 200:
+            logger.error("Remote contract verification failed.")
+            verification_result.sending_results_to_soda_cloud_failed = True
+            return verification_result
+
+        if not scan_id:
+            logger.warning("Did not receive a Scan ID from Soda Cloud")
+            return verification_result
```

Comment:
There's a quite few `return` scenarios here, with quite a lot of logic/setup/datastructures in between, would it make sense to refactor these bits out into separate functions so that the overall flow here is more intuitive to read? e.g. verify contract command dict setup+getting a response could be a candidate that would become a one-liner here, the logging logic below as well, wdyt?